[BUG] Can't start application with eagerInitSingletons turned on

### Expected Behavior

The application should start without problems when enabled `eagerInitSingletons`

### Actual Behaviour

Now, when launching the application, an error crashes:

```
Message: No bean of type [io.micronaut.http.client.HttpClientRegistry<io.micronaut.http.client.HttpClient>] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
Path Taken: new HttpClientIntroductionAdvice([HttpClientRegistry clientFactory],JsonMediaTypeCodec jsonMediaTypeCodec,List transformers,HttpClientBinderRegistry binderRegistry,ConversionService conversionService)
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [io.micronaut.http.client.interceptor.HttpClientIntroductionAdvice] could not be loaded: Failed to inject value for parameter [clientFactory] of class: io.micronaut.http.client.interceptor.HttpClientIntroductionAdvice

Message: No bean of type [io.micronaut.http.client.HttpClientRegistry<io.micronaut.http.client.HttpClient>] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
Path Taken: new HttpClientIntroductionAdvice([HttpClientRegistry clientFactory],JsonMediaTypeCodec jsonMediaTypeCodec,List transformers,HttpClientBinderRegistry binderRegistry,ConversionService conversionService)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1943)
	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:236)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3461)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:329)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:182)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:72)
	at com.micronaut.bug.Application.main(Application.java:15)
Caused by: io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [clientFactory] of class: io.micronaut.http.client.interceptor.HttpClientIntroductionAdvice

Message: No bean of type [io.micronaut.http.client.HttpClientRegistry<io.micronaut.http.client.HttpClient>] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
Path Taken: new HttpClientIntroductionAdvice([HttpClientRegistry clientFactory],JsonMediaTypeCodec jsonMediaTypeCodec,List transformers,HttpClientBinderRegistry binderRegistry,ConversionService conversionService)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:1588)
	at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForConstructorArgument(AbstractInitializableBeanDefinition.java:1044)
	at io.micronaut.http.client.interceptor.$HttpClientIntroductionAdvice$Definition.build(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2341)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3289)
	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:2669)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1937)
	... 6 common frames omitted
Caused by: io.micronaut.context.exceptions.NoSuchBeanException: No bean of type [io.micronaut.http.client.HttpClientRegistry<io.micronaut.http.client.HttpClient>] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2800)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1643)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:1566)
	... 12 common frames omitted

```

### Steps To Reproduce

Run example application

### Environment Information

OS: Windows 10
JDK Version: 17

### Example Application

https://github.com/altro3/micronaut3-bug

### Version

3.1.0

--- a/http-client-core/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java
+++ b/http-client-core/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java
@@ -110,8 +110,6 @@ public class HttpClientIntroductionAdvice implements MethodInterceptor<Object, O
      */
     private static final MediaType[] DEFAULT_ACCEPT_TYPES = {MediaType.APPLICATION_JSON_TYPE};
 
-    private static final int HEADERS_INITIAL_CAPACITY = 3;
-    private static final int ATTRIBUTES_INITIAL_CAPACITY = 1;
     private final List<ReactiveClientResultTransformer> transformers;
     private final HttpClientBinderRegistry binderRegistry;
     private final JsonMediaTypeCodec jsonMediaTypeCodec;

--- a/http-client-core/src/main/java/io/micronaut/http/client/package-info.java
+++ b/http-client-core/src/main/java/io/micronaut/http/client/package-info.java
@@ -13,10 +13,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 /**
- * Reactive transformers.
- *
- * @author Graeme Rocher
- * @since 1.0
+ * Core HTTP Client Library.
  */
-package io.micronaut.http.client.reactor;
+@Configuration
+@Requires(beans = HttpClientRegistry.class)
+package io.micronaut.http.client;
+
+import io.micronaut.context.annotation.Configuration;
+import io.micronaut.context.annotation.Requires;

--- a/http-client-core/src/main/java/io/micronaut/http/client/package-info.java
+++ b/http-client-core/src/main/java/io/micronaut/http/client/package-info.java
@@ -13,7 +13,6 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 /**
  * Core HTTP Client Library.
  */

--- a/http-client-core/src/test/groovy/io/micronaut/http/client/EagerInitSpec.groovy
+++ b/http-client-core/src/test/groovy/io/micronaut/http/client/EagerInitSpec.groovy
@@ -1,0 +1,15 @@
+package io.micronaut.http.client
+
+import io.micronaut.context.ApplicationContext
+import spock.lang.Specification
+
+class EagerInitSpec extends Specification {
+
+    void "test eager init singletons"() {
+        when:
+        ApplicationContext.builder().eagerInitSingletons(true).start()
+
+        then:
+        noExceptionThrown()
+    }
+}

